dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,553 @@
+                WebAPILocator.getHostWebAPI(),
+                APILocator.getAppsAPI(),
+                new SamlWebUtils(),
+                DotSamlFactory.getInstance().identityProviderConfigurationFactory(),
+                DotSamlFactory.getInstance().samlConfigurationService(),
+                DotSamlFactory.getInstance().samlAuthenticationService());
+    }
+
+    public SamlWebInterceptor(final Encryptor       encryptor,
+            final LoginServiceAPI loginService,
+            final UserAPI         userAPI,
+            final HostWebAPI      hostWebAPI,
+            final AppsAPI         appsAPI,
+            final SamlWebUtils    samlWebUtils,
+            final IdentityProviderConfigurationFactory identityProviderConfigurationFactory,
+            final SamlConfigurationService             samlConfigurationService,
+            final SamlAuthenticationService            samlAuthenticationService) {
+
+        this.encryptor    = encryptor;
+        this.loginService = loginService;
+        this.userAPI      = userAPI;
+        this.hostWebAPI   = hostWebAPI;
+        this.appsAPI      = appsAPI;
+        this.samlWebUtils = samlWebUtils;
+        this.identityProviderConfigurationFactory = identityProviderConfigurationFactory;
+        this.samlConfigurationService             = samlConfigurationService;
+        this.samlAuthenticationService            = samlAuthenticationService;
+    }
+
+    @Override
+    public Result intercept(final HttpServletRequest request,
+                            final HttpServletResponse response) throws IOException {
+
+        HttpSession session = request.getSession(false);
+
+        if (this.samlWebUtils.isByPass(request, session)) {
+
+            Logger.info(this, ()->"Using SAML by pass");
+            return Result.NEXT;
+        }
+
+        try {
+
+            if (null != session && this.isAnySamlConfigurated()) {
+
+                final Host host = hostWebAPI.getCurrentHostNoThrow(request);
+                final IdentityProviderConfiguration identityProviderConfiguration = // gets the SAML Configuration for this site.
+                        this.identityProviderConfigurationFactory.findIdentityProviderConfigurationById(
+                                host.getIdentifier());
+
+                // If idpConfig is null, means this site does not need SAML processing
+                if (null != identityProviderConfiguration && identityProviderConfiguration.isEnabled()) { // SAML is configurated, so continue
+
+                    // check if there is any exception filter path, to avoid to canApply all the logic.
+                    if (!this.checkAccessFilters(request.getRequestURI(), this.getAccessFilterArray(identityProviderConfiguration))
+                            && this.checkIncludePath(request.getRequestURI(), this.getIncludePathArray(identityProviderConfiguration))) {
+
+                        if (null == session || this.samlWebUtils.isNotLogged(request, session)) {
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)